src/boot/efi/util.c

```diff
@@ -769,6 +769,21 @@ EFI_STATUS make_file_device_path(EFI_HANDLE device, const char16_t *file, EFI_DE
         return EFI_SUCCESS;
 }
 
+EFI_STATUS device_path_to_text(const EFI_DEVICE_PATH *dp, char16_t **ret_str) {
+        EFI_DEVICE_PATH_TO_TEXT_PROTOCOL *text;
+        EFI_STATUS err;
+
+        assert(dp);
+        assert(ret_str);
+
+        err = BS->LocateProtocol(&(EFI_GUID) EFI_DEVICE_PATH_TO_TEXT_PROTOCOL_GUID, NULL, (void **) &text);
+        if (err != EFI_SUCCESS)
+                return err;
+
+        *ret_str = text->ConvertDevicePathToText(dp, true, true);
```

Comment:
Please do not initialize `ret_str` on failure.

(Not a matter, but I'd like to rename `text` -> `p` or something, as `text` indicates it is a string. Also, this function has single return value. So, how about to also rename `ret_str` -> `ret`?)